File: src/main/java/io/tatum/model/response/common/TxHash.java
package io.tatum.model.response.common;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
public class TxHash {
    /**
     * TX hash of successful transaction.
     * @type {string}
     * @memberof TxHash
     */
    private String txId;

    /**
     * Whethet withdrawal was completed in Tatum's internal ledger. If not, it must be done manually.
     * @type {boolean}
     * @memberof TxHash
     */
    private boolean completed;
}

File: src/main/kotlin/io/tatum/model/response/common/TxHash.kt
package io.tatum.model.response.common

/**
 * @property id ID of withdrawal. If transaction is not valid in blockchain, use this id to cancel withdrawal.
 * @property txId TX hash of successful transaction.
 * @property completed Whethet withdrawal was completed in Tatum's internal ledger. If not, it must be done manually.
 */
class TxHash(
    val id: String,
    val txId: String,
    val completed: Boolean,
)
